File: app/src/main/java/com/martinlaizg/geofind/views/adapter/PlayListAdapter.java
package com.martinlaizg.geofind.views.adapter;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.card.MaterialCardView;
import com.martinlaizg.geofind.R;
import com.martinlaizg.geofind.data.access.database.entities.Play;
import com.martinlaizg.geofind.data.access.database.entities.Tour;
import com.martinlaizg.geofind.views.fragment.single.TourFragment;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PlayListAdapter
		extends RecyclerView.Adapter<PlayListAdapter.PlaysViewHolder> {

	private List<Play> plays = new ArrayList<>();
	private Context context;

	@NonNull
	@Override
	public PlaysViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
		context = viewGroup.getContext();
		View view = LayoutInflater.from(viewGroup.getContext())
				.inflate(R.layout.fragment_play_item, viewGroup, false);
		return new PlaysViewHolder(view);
	}

	@Override
	public void onBindViewHolder(@NonNull PlaysViewHolder holder, final int i) {
		final Play play = plays.get(i);
		Tour t = play.getTour();
		holder.tour_name.setText(t.getName());
		holder.tour_creator.setText(t.getCreator().getUsername());
		holder.tour_description.setText(t.getDescription());

		if(t.getImage() != null && !t.getImage().isEmpty()) {
			Picasso.with(context).load(t.getImage()).into(holder.tour_image);
		} else {
			holder.tour_image.setImageResource(R.drawable.default_map_image);
		}
		int completed = play.getPlaces().size();
		int numPlaces = t.getPlaces().size();
		float progress = completed / (float) numPlaces * 100;
		holder.tour_progress.setProgress((int) progress, true);
		holder.tour_progress_text.setText(context.getString(R.string.div, completed, numPlaces));

		Bundle b = new Bundle();
		b.putInt(TourFragment.TOUR_ID, plays.get(i).getId());
		holder.tour_card
				.setOnClickListener(v -> Navigation.findNavController(v).navigate(R.id.toTour, b));
	}

	@Override
	public int getItemCount() {
		return plays.size();
	}

	public void setPlays(List<Play> plays) {
		this.plays = plays;
		notifyDataSetChanged();
	}

	class PlaysViewHolder
			extends RecyclerView.ViewHolder {

		@BindView(R.id.tour_image)
		ImageView tour_image;
		@BindView(R.id.tour_name)
		TextView tour_name;
		@BindView(R.id.tour_creator)
		TextView tour_creator;
		@BindView(R.id.tour_description)
		TextView tour_description;
		@BindView(R.id.tour_card)
		MaterialCardView tour_card;

		@BindView(R.id.tour_progress)
		ProgressBar tour_progress;
		@BindView(R.id.tour_progress_text)
		TextView tour_progress_text;

		PlaysViewHolder(@NonNull View itemView) {
			super(itemView);
			ButterKnife.bind(this, itemView);
		}
	}

}

File: app/src/main/java/com/martinlaizg/geofind/views/adapter/PlayListAdapter.kt
package com.martinlaizg.geofind.views.adapter

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.card.MaterialCardView
import com.martinlaizg.geofind.R
import com.martinlaizg.geofind.data.access.database.entities.Play
import com.martinlaizg.geofind.views.adapter.PlayListAdapter.PlaysViewHolder
import com.martinlaizg.geofind.views.fragment.single.TourFragment
import java.util.*

class PlayListAdapter : RecyclerView.Adapter<PlaysViewHolder>() {

	var plays: List<Play> = ArrayList()
		set(value) {
			notifyDataSetChanged()
			field = value
		}

	private var context: Context? = null

	override fun onBindViewHolder(holder: PlaysViewHolder, i: Int) = holder.bind(plays[i])
	override fun getItemCount(): Int = plays.size
	override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): PlaysViewHolder {
		context = viewGroup.context
		val view = LayoutInflater.from(viewGroup.context)
				.inflate(R.layout.fragment_play_item, viewGroup, false)
		return PlaysViewHolder(view)
	}

	inner class PlaysViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
		fun bind(play: Play) {
			val t = play.tour
			tourName.text = t!!.name
			tourCreator.text = t.creator!!.username
			tourDescription.text = t.description
			if (t.image != null && t.image!!.isNotEmpty()) {
				TODO("load image")
			}
			val completed = play.places.size
			val numPlaces = t.places.size
			val progress = completed / numPlaces.toFloat() * 100
			tourProgress.setProgress(progress.toInt(), true)
			tourProgressText.text = context!!.getString(R.string.div, completed, numPlaces)
			val b = Bundle()
			b.putInt(TourFragment.TOUR_ID, t.id)
			tourCard.setOnClickListener { v: View? -> Navigation.findNavController(v!!).navigate(R.id.toTour, b) }
		}

		var tourImage: ImageView = itemView.findViewById(R.id.tour_image)
		private var tourName: TextView = itemView.findViewById(R.id.tour_name)
		private var tourCreator: TextView = itemView.findViewById(R.id.tour_creator)
		private var tourDescription: TextView = itemView.findViewById(R.id.tour_description)
		private var tourCard: MaterialCardView = itemView.findViewById(R.id.tour_card)
		private var tourProgress: ProgressBar = itemView.findViewById(R.id.tour_progress)
		private var tourProgressText: TextView = itemView.findViewById(R.id.tour_progress_text)
	}
}